### PR TITLE
280 user controller testing

### LINKED ISSUE #280: UserController.java Testing

### PR DESCRIPTION
**Pull request by Juanan-design**
Resolves #280 

# Description

Added unit test cases to userController.java

### CODE REVIEW

File: backend/src/test/java/controller/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserControllerTest {
+
+    @InjectMocks
+    private UserController userController;
+
+    @Mock
+    private UserService userService;
+
```
Comment (mpk467) [inline backend/src/test/java/controller/UserControllerTest.java:43]: it is also possible to mock httpsession with
` @Mock
private HttpSession httpSession;`

Comment (mpk467) [reply backend/src/test/java/controller/UserControllerTest.java:43]: check out WatchControllerTest.java https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-3/blob/development/backend/src/test/java/controller/WatchControllerTest.java

**Review by ralphgregorio — APPROVED** [review]
👍

File: backend/src/test/java/controller/UserControllerTest.java (hunk 1 above)
Comment (kamalzrein) [reply backend/src/test/java/controller/UserControllerTest.java:43]: Looks good ! good work!

**Review by kamalzrein — APPROVED** [review]
Tests look good!

**Pull request merged**